sizeadjustingtextview/src/main/java/com/omitneedlesscode/sizeadjustingtextview/SizeAdjustingTextView.java

```diff
@@ -0,0 +1,172 @@
+package com.omitneedlesscode.sizeadjustingtextview;
+
+import ohos.agp.components.AttrSet;
+import ohos.agp.components.Text;
+import ohos.agp.render.Paint;
+import ohos.agp.text.SimpleTextLayout;
+import ohos.agp.utils.Rect;
+import ohos.agp.utils.TextAlignment;
+import ohos.app.Context;
+import ohos.hiviewdfx.HiLog;
+import ohos.hiviewdfx.HiLogLabel;
+
+/**
+ * SizeAdjustingTextView is a TextView that adjusts text size
+ * depending if or if not it will fit in the view bounds.
+ * Copyright (C) 2014 Elliott Chenger
+ * <p>
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * <p>
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * @author Elliott Chenger
+ * @author Chase Colburn
+ * @since Jan 18, 2014
+ * @since Apr 4, 2011
+ */
+public class SizeAdjustingTextView extends Text implements Text.TextObserver {
+    public static final HiLogLabel HI_LOG_LABEL = new HiLogLabel(0, 0, "SizeAdjustingTextView");
+
+    Paint mTextPaint;
+
+    SimpleTextLayout mTextLayout;
```

Comment:
Paint mTextPaint;
SimpleTextLayout mTextLayout;

Can be made as private as they need not be accessed outside the file.

Reply:
Done.